### PR TITLE
Refactors server configuration README

### PR DESCRIPTION
**Pull request by codefromthecrypt**
This refactors the server README to remove redundant items about how we
map env to properties. This also adds an example of what property file
Spring Boot will read by default for overrides: `zipkin-server.properties`.

The motivation for this is modules who have other configuration, and a
number of questions over the last years about property files.

### CODE REVIEW

**Comment by codefromthecrypt** [comment]
cc @openzipkin/core as I think everyone should know that `$PWD/zipkin-server.properties` is magically read. When folks insist on properties files, this should allow a consistent answer, while simultaneously transferring support burden to those who choose to do that.

**Review by shakuzen — APPROVED** [review]
LGTM

**Comment by codefromthecrypt** [comment]
Latest heading rewords things to hopefully avoid custom servers. I'd forgotten we firstly called Zipkin a Spring Boot server, which perhaps led to people thinking they can just customize it. While we do rely on Spring boot, most things we do are custom or Armeria specific. cc @trustin @minwoox

File: zipkin-server/README.md [hunk 1]
```diff
@@ -100,26 +118,50 @@ counter.zipkin_collector.spans_dropped.$transport | cumulative spans dropped; re
...
-[yaml configuration](src/main/resources/zipkin-server-shared.yml) binds the following environment variables to spring properties:
-
-Variable | Property | Description
---- | --- | ---
-SELF_TRACING_ENABLED | zipkin.self-tracing.enabled | Set to true to enable self-tracing. Defaults to false
-SELF_TRACING_SAMPLE_RATE`: Percentage of self-traces to retain, defaults to always sample (1.0).
-SELF_TRACING_FLUSH_INTERVAL | zipkin.self-tracing.flush-interval | Interval in seconds to flush self-tracing data to storage. Defaults to 1
+Here are the top-level configuration of Zipkin:
+* `QUERY_PORT`: Listen port for the http api and web ui; Defaults to 9411
+* `QUERY_ENABLED`: `false` disables the query api and UI assets. Search
```
Comment (anuraaga) [inline zipkin-server/README.md]: I don't understand what this sentence is saying, Search may be disabled for the storage backend. Is it possible to rephrase it?

File: zipkin-server/README.md [hunk 2]
```diff
@@ -100,26 +118,50 @@ counter.zipkin_collector.spans_dropped.$transport | cumulative spans dropped; re
...
+may also be disabled for the storage backend if it is not needed;
+Defaults to true
+* `SEARCH_ENABLED`: `false` disables trace search requests on the storage
```
Comment (anuraaga) [inline zipkin-server/README.md]: I think this relates to my above comment but if query is disabled isn't search always disabled? Do we need to mention storage backend?

**Review by anuraaga — APPROVED** [review]

File: zipkin-server/README.md (hunk 1 above)
Comment (codefromthecrypt) [reply zipkin-server/README.md]: this content was just moved up. I can address this now, if you think it is relevant to..

File: zipkin-server/README.md (hunk 2 above)
Comment (codefromthecrypt) [reply zipkin-server/README.md]: this content was just moved up. I can address this now, if you think it is relevant to..

**Comment by anuraaga** [comment]
Yeah don't have to but figured it's a good chance to

File: zipkin-server/README.md (hunk 1 above)
Comment (codefromthecrypt) [reply zipkin-server/README.md]: agree it is awkward.. I'll do this here as it shouldn't be too much chatter to find good text.

**Review by anuraaga — APPROVED** [review]
Thanks for tweaking!

File: zipkin-server/README.md [hunk 3]
```diff
@@ -1,9 +1,12 @@
...
 
-Span storage and collectors are configurable. By default, storage is
-in-memory, the http collector (POST /api/v2/spans endpoint) is enabled,
-and the server listens on port 9411.
+Span storage and collectors are [configurable](#configuration). By default, storage is in-memory,
+the http collector (POST /api/v2/spans endpoint) is enabled, and the server listens on port 9411.
+
+Zipkin Server is implemented with [Armeria](https://github.com/line/armeria) and uses [Spring Boot](http://docs.spring.io/spring-boot/docs/current/reference/htmlsingle/)
+internally for configuration. It should not be considered a normal Spring Boot application: custom
+servers are not supported.
```
Comment (jcchavezs) [inline zipkin-server/README.md]: I would bold **custom servers are not supported.**

File: zipkin-server/README.md [hunk 4]
```diff
@@ -100,26 +118,50 @@ counter.zipkin_collector.spans_dropped.$transport | cumulative spans dropped; re
...
+We support ENV variable configuration, such as `STORAGE_TYPE=cassandra3`, as they are familiar to
+administrators and easy to use in runtime environments such as Docker.
 
-Variable | Property | Description
---- | --- | ---
-SELF_TRACING_ENABLED | zipkin.self-tracing.enabled | Set to true to enable self-tracing. Defaults to false
-SELF_TRACING_SAMPLE_RATE`: Percentage of self-traces to retain, defaults to always sample (1.0).
-SELF_TRACING_FLUSH_INTERVAL | zipkin.self-tracing.flush-interval | Interval in seconds to flush self-tracing data to storage. Defaults to 1
+Here are the top-level configuration of Zipkin:
+* `QUERY_PORT`: Listen port for the http api and web ui; Defaults to 9411
```
Comment (jcchavezs) [inline zipkin-server/README.md]: nit: `HTTP API` and `Web UI` to be consistent with the next paragraph?

File: zipkin-server/README.md [hunk 5]
```diff
@@ -459,6 +477,19 @@ $ COLLECTOR_GRPC_ENABLED=true java -jar zipkin.jar
 
 As this service is experimental, it is not recommended to run this in production environments.
 
+## Self-Tracing
+Self tracing exists to help troubleshoot performance of the zipkin-server. Production deployments
+who enable self-tracing should lower the sample rate from 1.0 (100%) to a much smaller rate, like
+0.001 (0.1% or 1 out of 1000).
+
+When `zipkin.self-tracing.enabled=true`, Zipkin will self-trace calls to the api.
```
Comment (jcchavezs) [inline zipkin-server/README.md]: How about specifying what is the name of the service or how to find such traces?

Comment (codefromthecrypt) [reply zipkin-server/README.md]: done!

File: zipkin-server/README.md (hunk 4 above)
Comment (codefromthecrypt) [reply zipkin-server/README.md]: done everywhere

File: zipkin-server/README.md (hunk 3 above)
Comment (codefromthecrypt) [reply zipkin-server/README.md]: **good point**

Comment (minwoox) [reply zipkin-server/README.md]: :rofl:

**Comment by codefromthecrypt** [comment]
build broke due to travis git still having issues, not the actual change here. #2879

merging and thanks all for the feedback!

**Pull request merged**